src/components/views/elements/EffectsOverlay.tsx

```diff
@@ -30,15 +30,15 @@ const EffectsOverlay: FunctionComponent<IProps> = ({ roomWidth }) => {
     const canvasRef = useRef<HTMLCanvasElement>(null);
     const effectsRef = useRef<Map<string, ICanvasEffect>>(new Map<string, ICanvasEffect>());
 
-    const lazyLoadEffectModule = async (name: string): Promise<ICanvasEffect> => {
+    const lazyLoadEffectModule = async (name: string): Promise<ICanvasEffect | null> => {
         if (!name) return null;
         let effect: ICanvasEffect | null = effectsRef.current.get(name) || null;
         if (effect === null) {
             const options = CHAT_EFFECTS.find((e) => e.command === name)?.options;
             try {
                 const { default: Effect } = await import(`../../../effects/${name}`);
                 effect = new Effect(options);
-                effectsRef.current.set(name, effect);
+                effectsRef.current.set(name, effect!);
```

Comment:
`effect` is created one line before. Does typescript really not know that `effect` exists here?

Reply:
Effect is untyped due to `await import(`../../../effects/${name}`);`